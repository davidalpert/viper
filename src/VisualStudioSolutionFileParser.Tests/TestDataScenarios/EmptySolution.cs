using System.Linq;
using ApprovalTests;
using NUnit.Framework;
using VisualStudioSolutionFileParser.AST;
using VisualStudioSolutionFileParser.Tests.Helpers;

namespace VisualStudioSolutionFileParser.Tests.TestDataScenarios
{
    public class EmptySolution
    {
        public SolutionFile ParseTestData()
        {
            var input = ManifestResourceHelper.ExtractManifestResourceToString("TestData.EmptySolution.sln");
            return Parser.Parse(input);
        }

        [Test]
        public void CanParse()
        {
            var result = ParseTestData();

            Assert.NotNull(result);
        }

        [Test]
        [Ignore("deferred until I settle on a method to pretty print the AST")]
        public void CanPrint()
        {
            var result = ParseTestData();

            Approvals.Verify(result);
        }
    }
}