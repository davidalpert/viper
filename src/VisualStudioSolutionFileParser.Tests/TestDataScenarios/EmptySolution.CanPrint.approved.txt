Visual Studio 2012 : 12.0